assets/app/scripts/services/api.js

```diff
@@ -0,0 +1,255 @@
+  return this;
+}
+// toString() includes the group and version information if present
+ResourceGroupVersion.prototype.toString = function() {
+  var s = this.resource;
+  if (this.group)   { s += "/" + this.group;   }
+  if (this.version) { s += "/" + this.version; }
+  return s;
+};
+// primaryResource() returns the resource with any subresources removed
+ResourceGroupVersion.prototype.primaryResource = function() {
+  var i = this.resource.indexOf('/');
+  if (i === -1) { return this.resource; }
+  return this.resource.substring(0,i);
+};
+// subresources() returns a (possibly empty) list of subresource segments
+ResourceGroupVersion.prototype.subresources = function() {
+  var i = this.resource.indexOf('/');
+  if (i === -1) { return []; }
+  var resources = this.resource.split("/");
+  resources.shift();
+  return resources;
+};
+// equals() returns true if the given resource, group, and version match.
+// If omitted, group and version are not compared.
+ResourceGroupVersion.prototype.equals = function(resource, group, version) {
+  if (this.resource !== resource) { return false; }
+  if (arguments.length === 1)     { return true;  }
+  if (this.group !== group)       { return false; }
+  if (arguments.length === 2)     { return true;  }
+  if (this.version !== version)   { return false; }
+  return true;
+};
+
+
+angular.module('openshiftConsole')
+.factory('APIService', function(API_CFG, APIS_CFG, Logger) {
+  
+  // Set the default api versions the console will use if otherwise unspecified
+  var defaultVersion = {
+    "":           "v1",
+    "extensions": "v1beta1"
+  };
+
+  // toResourceGroupVersion() returns a ResourceGroupVersion.
+  // If resource is already a ResourceGroupVersion, returns itself.
+  //
+  // if r is a string, the empty group and default version for the empty group are assumed.
+  //
+  // if r is an object, the resource, group, and version attributes are read.
+  // a missing group attribute defaults to the legacy group.
+  // a missing version attribute defaults to the default version for the group, or undefined if the group is unknown.
+  //
+  // if r is already a ResourceGroupVersion, it is returned as-is
+  var toResourceGroupVersion = function(r) {
+    if (r instanceof ResourceGroupVersion) {
+      return r;
+    }
+    var resource, group, version;
+    if (angular.isString(r)) {
+      resource = normalizeResource(r);
+      group = '';
+      version = defaultVersion[group];
+    } else if (r && r.resource) {
+      resource = normalizeResource(r.resource);
+      group = r.group || '';
+      version = r.version || defaultVersion[group];
+    }
+    return new ResourceGroupVersion(resource, group, version);
+  };
+  // toResourceGroupVersion() is identical to toResourceGroupVersion() but warns if a non-ResourceGroupVersion is seen.
+  // TODO: remove after ensuring all code paths that call this already have a ResourceGroupVersion.
+  var toResourceGroupVersionWithWarning = function(r) {
```

Comment:
I haven't hit the warning. I've tested every page and performed every action I can think of.